src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -369,6 +371,16 @@ internal void TrackReplacedEventHandlerId(ulong oldEventHandlerId, ulong newEven
             _eventHandlerIdReplacements.Add(oldEventHandlerId, newEventHandlerId);
         }
 
+        /// <summary>
+        /// Gets the <see cref="IJSRuntime"/> service from the <see cref="Renderer"/>'s service provider.
+        /// </summary>
+        /// <remarks>
+        /// The returned <see cref="IJSRuntime"/> may be <c>null</c> if the <see cref="Renderer"/>'s service
+        /// provider does not have one.
+        /// </remarks>
+        internal IJSRuntime GetJSRuntime()
```

Comment:
```suggestion
        internal IJSRuntime JSRuntime { get; } = _serviceProvider.GetRequiredService<IJSRuntime>();
```

if this doesn't work, you could assign the property in the constructor.

Reply:
Based on Steve's comment, I guess this should continue to use `GetService` in the event there are environments this isn't supported.